trinity/main.py

```diff
@@ -321,9 +314,9 @@ def kill_trinity_gracefully(logger: logging.Logger,
 
 @setup_cprofiler('run_database_process')
 @with_queued_logging
-def run_database_process(trinity_config: TrinityConfig, db_class: Type[BaseDB]) -> None:
+def run_database_process(trinity_config: TrinityConfig) -> None:
     with trinity_config.process_id_file('database'):
-        base_db = db_class(db_path=trinity_config.database_dir)
+        base_db = trinity_config.db_class(db_path=trinity_config.database_dir)
```

Comment:
I think rocks and level use names that prevent filepath collisions, but worth double checking.

Reply:
I hadn't gotten this far but yes, we should be very sure that we cannot re-use a database directory that was previously used with another backend.  

I thought about fixing this by namespacing the directory based on the backend but... then we could get into a situation where the user switches the backend midstream and effectively has two versions of the same chain.

So maybe just a small file we place somewhere to be able to check what database is being used.

Reply:
TODO: make sure leveldb and rocksdb databases cannot use the same root directory